dao/mongodb.rya/src/main/java/org/apache/rya/mongodb/document/util/DocumentVisibilityUtil.java

```diff
@@ -0,0 +1,316 @@
+     * @param expression the expression byte array.
+     * @return the multidimensional array representation of the boolean
+     * expression.
+     */
+    public static Object[] toMultidimensionalArray(final Node node, final byte[] expression) {
+        checkNotNull(node);
+        final List<Object> array = new ArrayList<>();
+
+        if (node.getChildren().isEmpty() && node.getType() == NodeType.TERM) {
+            final String data = getTermNodeData(node);
+            array.add(data);
+        }
+
+        log.trace("Children size: " + node.getChildren().size() + " Type: " + node.getType());
+        for (final Node child : node.getChildren()) {
+            switch (child.getType()) {
+                case EMPTY:
+                case TERM:
+                    String data;
+                    if (child.getType() == NodeType.TERM) {
+                        data = getTermNodeData(child);
+                    } else {
+                        data = "";
+                    }
+                    if (node.getType() == NodeType.OR) {
+                        array.add(Lists.newArrayList(data).toArray(new Object[0]));
+                    } else {
+                        array.add(data);
+                    }
+                    break;
+                case OR:
+                case AND:
+                    array.add(toMultidimensionalArray(child, expression));
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        return array.toArray(new Object[0]);
+    }
+
+    public static String nodeToBooleanString(final Node node) {
+        boolean isFirst = true;
+        final StringBuilder sb = new StringBuilder();
+        if (node.getType() == NodeType.TERM) {
+            final String data = getTermNodeData(node);
+            sb.append(data);
+        }
+        if (node.getType() == NodeType.AND) {
+            sb.append("(");
+        }
+        for (final Node child : node.getChildren()) {
+            if (isFirst) {
+                isFirst = false;
+            } else {
+                if (node.getType() == NodeType.OR) {
+                    sb.append("|");
+                } else if (node.getType() == NodeType.AND) {
+                    sb.append("&");
+                }
+            }
+            switch (child.getType()) {
+                case EMPTY:
+                    sb.append("");
+                    break;
+                case TERM:
+                    final String data = getTermNodeData(child);
+                    sb.append(data);
+                    break;
+                case OR:
+                    sb.append("(");
+                    sb.append(nodeToBooleanString(child));
+                    sb.append(")");
+                    break;
+                case AND:
+                    sb.append(nodeToBooleanString(child));
+                    break;
+                default:
+                    break;
```

Comment:
default shouldn't be possible.  I meant to add an error in there if it did reach it so I'll put that in now.